### ISSUE TITLE
`cargo watch -x run` after some saves turns the terminal into a mess.

### ISSUE THREAD

**Issue by h4rldev** [watchexec/cargo-watch#269]
### Issue
It sometimes essentially locks the entire terminal and throws the text everywhere on save.
Might it be a conflict with the "menu" or something?

### Screenshot
![image](https://github.com/watchexec/cargo-watch/assets/98224660/0dc239c1-5705-487b-a0a8-6365a0b2d454)

**Comment by passcod** [comment]
yeah, that's your program not resetting the screen gracefully on restart. try doing

```
cargo watch -s 'reset; cargo run'
```

**Comment by h4rldev** [comment]
Any way for me to implement that in my code?

**Comment by passcod** [comment]
sure, hook to sigint/ctrlc and cleanup state, or as a last resort, forcibly reset the screen. you can use the clearscreen crate for this, or your terminal manipulation library may have a facility

**Comment by h4rldev** [comment]
After looking up on this I seem to not find a way to catch the termination signal of cargo-watch and reset the screen

**Comment by h4rldev** [comment]
And my program seems to start thinking I pressed enter twice

**Comment by passcod** [comment]
You're right, I keep getting confused between the current states of cargo watch and watchexec. There's no easy way to catch that signal at the moment. If you use watchexec directly instead (a bit less nice for cargo projects), you'll get a SIGTERM by default, and can customise that too, and you can use `--clear=reset` as a simpler pattern than using `tput reset`.

**Comment by h4rldev** [comment]
Any plans to implement that for cargo-watch?

**Comment by passcod** [comment]
definitely. currently waiting for/working on:
- ignores to stabilise (ie for any more bugs to come out after the latest watchexec release)
- a fix for the startup behaviour (next priority)
- filter programs (pending a third party crate release)
- proper multi-command support (currently in review)

once the first two are done in watchexec,  i'll consider it stable enough to start porting cargo watch, at which point i'll aim for cli flags feature parity.

tl;dr: not immediately but it's in sight

**Comment by passcod** [comment]
definitely. currently waiting for/working on:
- ignores to stabilise (ie for any more bugs to come out after the latest watchexec release)
- a fix for the startup behaviour (next priority)
- filter programs (pending a third party crate release)
- proper multi-command support (currently in review)

once the first two are done in watchexec,  i'll consider it stable enough to start porting cargo watch, at which point i'll aim for cli flags feature parity.

tl;dr: not immediately but it's in sight
